### PR TITLE
refactor(bittensor): use WalletCore AnyAddress for SS58 encode/decode

### LINKED ISSUE #5667: Replace BittensorHelper's hand-rolled SS58 encode/decode with WalletCore AnyAddress
## Description

`BittensorHelper` hand-rolls SS58 address encoding and decoding — base58, the blake2b-512 `"SS58PRE"` checksum, and the 1-byte/2-byte prefix length branch — in ~80 lines that WalletCore already implements natively.

**File:** `data/src/main/kotlin/com/vultisig/wallet/data/chains/helpers/BittensorHelper.kt`

- `ss58Encode` (line 238) — base58 of `prefix ++ pubkey ++ blake2b-512("SS58PRE" ++ payload)[0..1]`
- `ss58Decode` (line 275) — inverse, plus checksum verification
- Supporting private helpers: `base58Encode` (249), `base58Decode` (307), `BASE58_ALPHABET` (305)

WalletCore 4.7.0 exposes the same operations on `wallet.core.jni.AnyAddress`:

- `AnyAddress(PublicKey, CoinType, int ss58Prefix)` → `description()` for encode
- `AnyAddress(String, CoinType, int ss58Prefix)` → `data()` for decode
- `AnyAddress.isValidSS58(String, CoinType, int ss58Prefix)` for validation

The repository already uses the third one — `ChainAccountAddressRepository.isValid()` validates Bittensor addresses with `AnyAddress.isValidSS58(address, CoinType.POLKADOT, 42)`. Only the encode/decode halves are hand-rolled.

## Why this is safe to swap

The hand-rolled path is not pure Kotlin — both `ss58Encode` and `ss58Decode` already call `wallet.core.jni.Hash.blake2b` (lines 244 and 298), so they already require the native library and already cannot run in a local JVM unit test. No unit test currently covers them. Replacing them with `AnyAddress` trades one JNI dependency for another and loses no test coverage.

## Proposed change

Keep both function names so the three call sites are untouched, and replace only the bodies:

```kotlin
fun ss58Encode(pubkey: ByteArray): String {
    require(pubkey.size == 32) { "SS58 encode requires 32-byte pubkey, got ${pubkey.size}" }
    return AnyAddress(PublicKey(pubkey, PublicKeyType.ED25519), CoinType.POLKADOT, SS58_PREFIX)
        .description()
}

fun ss58Decode(address: String): ByteArray =
    AnyAddress(address, CoinType.POLKADOT, SS58_PREFIX).data()
```

`base58Encode`, `base58Decode` and `BASE58_ALPHABET` then become dead and can go. `hexToBytes` (line 264) must stay — it is used by `ChainAccountAddressRepository`.

## Call sites

- `ChainAccountAddressRepository` — `ss58Encode` on the EdDSA address-derivation path. A `PublicKey(eddsaPubKey, ED25519)` is already constructed two lines above, so the encode path needs no new object.
- `BittensorApi.computeAccountStorageKey` — `ss58Decode` to recover the pubkey for the `System.Account` storage key.
- `BittensorHelper.buildCallData` (line 103) — `ss58Decode` on `keysignPayload.toAddress` to get the 32-byte `MultiAddress::Id` destination.

## Behaviour deltas to confirm

1. **`AnyAddress.data()` must return the 32-byte account id for Polkadot**, not a hash of it. This is the one thing that has to be verified on device before the change can be trusted — the JNI jar exposes only the shim, so the C++ semantics cannot be read from the dependency. A round-trip check against the current implementation's output settles it.

2. **Decode becomes stricter, in the safe direction.** The current `ss58Decode` slices off the prefix and verifies the checksum but never checks that the prefix equals 42, so a Polkadot `1…` address decodes without complaint. WalletCore validates the address against the prefix passed in. `isValid()` already gates user-entered addresses at prefix 42, so nothing legitimate is rejected; this closes the gap on the unguarded `buildCallData` path. Related: #3993.

3. **Exception type stays compatible.** `AnyAddress` throws `java.security.InvalidParameterException`, which extends `IllegalArgumentException` — the same type thrown today. The message text changes though (it loses `"Invalid SS58 checksum for address: …"`), so any UI surfacing that string should be checked.

4. **The explicit `SS58_PREFIX = 42` must be kept.** `CoinType.POLKADOT.ss58Prefix()` is 0; dropping the override would silently produce Polkadot `1…` addresses instead of Bittensor `5…` ones.

## Test plan

- Add an instrumented parity test under `data/src/androidTest` (the module already has `WalletCoreNative` for loading the native library) asserting the new path round-trips to the same bytes and the same string as the current implementation for a set of real TAO addresses. This is the only way to settle delta 1.
- Verify Bittensor address derivation for an existing vault is unchanged.
- Verify a TAO send: balance load (storage key path) and a signed extrinsic whose destination bytes match the previous encoder.

### PR DESCRIPTION
**Pull request by johnnyluo**
## Description

`BittensorHelper` hand-rolled SS58 address encoding/decoding — base58 conversion plus the blake2b-512 `SS58PRE` checksum — in ~80 lines that WalletCore's `AnyAddress` already implements natively (the repo already uses `AnyAddress.isValidSS58` for validation). This swaps only the `ss58Encode`/`ss58Decode` bodies for `AnyAddress` calls, keeps the explicit Bittensor SS58 prefix (42), and leaves all three call sites and function signatures untouched.

Fixes #5667

## Which feature is affected?
- [ ] Create vault ( Secure / Fast) - Please ensure you created a Secure vault & fast vault
- [x] Sending  - Bittensor address derivation and the SS58 destination decode used when building a transfer extrinsic both go through this path
- [ ] Swap - Please attach a tx link for the swap here
- [ ] New Chain / Chain related feature  -  Please attach tx link here

## Parity

- iOS: n/a — internal refactor of this Android module's hand-rolled codec, no behavior change to mirror
- Windows + extension: n/a — same
- SDK: n/a — same

## Checklist

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

## Screenshots (if applicable):

N/A — non-UI refactor.

## Additional context

One intentional behavior delta, called out in the issue: `ss58Decode` now rejects an address encoded with any SS58 prefix other than 42 (WalletCore validates the prefix), whereas the old decoder only verified the checksum and ignored the prefix's value. This closes a validation gap on the `buildCallData` path; `isValid()` already gated user-entered addresses at prefix 42 elsewhere.

Added `BittensorHelperSs58ParityTest` (instrumented, `data/src/androidTest`) which keeps the old hand-rolled algorithm as a reference oracle and checks the new `AnyAddress`-backed implementation against it for parity, including a round-trip and a leading-zero-byte edge case, plus a test for the stricter-prefix rejection.

**Not yet verified on-device**: I could not run the instrumented test in this environment (no `adb`/emulator, and WalletCore's native library is Android-only). `:data:compileDebugAndroidTestKotlin` and the full `:data:testDebugUnitTest` suite both pass, but the actual JNI-backed parity check needs to run on a device or CI before merge — please run `./gradlew :data:connectedDebugAndroidTest --tests "*BittensorHelperSs58ParityTest*"` on an emulator/device.

## Agent Metadata
- **Agent**: Claude Code
- **Issue**: #5667
- **Confidence**: medium — logic mirrors the issue's own analysis and compiles/tests clean, but the one risk flagged by the issue itself (whether `AnyAddress.data()` returns the raw account id) is unverified on-device
- **Needs human review for**: running the instrumented parity test on a device before merge

Co-Authored-By: Claude Sonnet 5 <noreply@anthropic.com>

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved Bittensor address encoding and decoding for greater compatibility and reliable validation.
  - Ensured addresses with incorrect network prefixes are rejected.

- **Tests**
  - Added coverage for random keys, round-trip conversions, reference compatibility, and invalid-prefix handling.

### CODE REVIEW

File: data/src/androidTest/kotlin/com/vultisig/wallet/data/chains/helpers/BittensorHelperSs58ParityTest.kt [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                n = quot
+            }
+            for (b in data) {
+                if (b == 0.toByte()) sb.append('1') else break
+            }
+            return sb.reverse().toString()
+        }
+    }
+
+    private fun randomPubkey(seed: Long): ByteArray {
```
Comment (rkokhatskyi) [inline data/src/androidTest/kotlin/com/vultisig/wallet/data/chains/helpers/BittensorHelperSs58ParityTest.kt]: `randomPubkey` feeds `ss58Encode` arbitrary 32 bytes, but WalletCore rejects anything that is not a compressed Edwards point and `AnyAddress` then returns an empty string instead of throwing, so an emulator run fails two of five with `seed 13 expected:<5GKEzYSSVkCDxEwW2QhiHqcR2QMBEBbaDUR2ctEsj1Ur5r3p> but was:<>`

File: app/src/androidTest/java/com/vultisig/wallet/data/chains/helpers/BittensorHelperSs58ParityTest.kt [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import java.math.BigInteger
+import java.security.SecureRandom
+import org.junit.Assert.assertArrayEquals
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertThrows
+import org.junit.Before
+import org.junit.Test
+
+/**
+ * Parity coverage for #5667: [BittensorHelper.ss58Encode]/[BittensorHelper.ss58Decode] now delegate
```
Comment (rkokhatskyi) [inline app/src/androidTest/java/com/vultisig/wallet/data/chains/helpers/BittensorHelperSs58ParityTest.kt:15]: CI runs only `./gradlew app:connectedDebugAndroidTest` at `.github/workflows/android.yml:160`, which never reaches the `:data` module, so this file is neither compiled nor executed in CI and the green instrumented-tests check on this PR covers none of it

**Review by rkokhatskyi — CHANGES_REQUESTED** [review]

**Review by aminsato — APPROVED** [review]
LGTM

File: data/src/main/kotlin/com/vultisig/wallet/data/chains/helpers/BittensorHelper.kt [hunk 3]
```diff
@@ -232,33 +234,16 @@ class BittensorHelper(private val vaultHexPublicKey: String) {
         }
 
         /**
-         * SS58 encode raw 32-byte pubkey with Bittensor prefix (42). Format: base58(prefix_byte ++
-         * pubkey ++ blake2b_checksum[0..2])
+         * SS58 encode raw 32-byte pubkey with Bittensor prefix (42), via WalletCore's AnyAddress.
          */
         fun ss58Encode(pubkey: ByteArray): String {
             require(pubkey.size == 32) { "SS58 encode requires 32-byte pubkey, got ${pubkey.size}" }
```
Comment (aminsato) [inline data/src/main/kotlin/com/vultisig/wallet/data/chains/helpers/BittensorHelper.kt:240]: Clean handoff to WalletCore — the reference test (BittensorHelperSs58ParityTest) pins the old implementation and verifies this produces identical results for 20 random keys, plus validates the prefix like it should.

**Review by rkokhatskyi — APPROVED** [review]
LGTM
